tools/data-conversion/src/main.rs

```diff
@@ -549,3 +627,369 @@ fn binary_to_fixed_bit_slice(
+    let (integer_part, fractional_part) = if decimal_position > 0 {
+        mantissa_str.split_at(decimal_position)
+    } else {
+        ("0", mantissa_str)
+    };
+
+    let result = if inter_rep.sign {
+        format!("{}.{}", integer_part, fractional_part)
+    } else {
+        format!("-{}.{}", integer_part, fractional_part)
+    };
+
+    if let Some(file) = filepath_send.as_mut() {
+        // Write string to the file
+        file.write_all(result.as_bytes())?;
+        file.write_all(b"\n")?;
+    } else {
+        io::stdout().write_all(result.as_bytes())?;
+        io::stdout().write_all(b"\n")?;
+    }
+
+    Ok(())
+}
+
+/// Converts a string representation of a fixed-point number to the intermediate representation.
+///
+/// This function takes a string slice representing a fixed-point number and an exponent value,
+/// determines the sign, constructs the mantissa, sets the exponent to the given value, and
+/// constructs an `IntermediateRepresentation` containing the sign, mantissa, and exponent.
+///
+/// # Arguments
+///
+/// * `fixed_string` - A string slice containing the fixed-point number to be converted.
+/// * `exp_int` - The exponent value for the fixed-point number.
+///
+/// # Returns
+///
+/// This function returns an `IntermediateRepresentation` containing the sign, mantissa,
+/// and exponent of the fixed-point number.
+///
+/// # Panics
+///
+/// This function will panic if the input string cannot be parsed as a number.
+fn fixed_to_intermediate(
+    fixed_string: &str,
+    exp_int: i32,
+) -> IntermediateRepresentation {
+    let sign = !fixed_string.starts_with('-');
+    let fixed_trimmed = fixed_string.trim_start_matches('-');
+
+    let mantissa_string = &format!("{fixed_trimmed}");
+
+    IntermediateRepresentation {
+        sign,
+        mantissa: BigUint::from_str(mantissa_string).expect("Invalid number"),
+        exponent: exp_int,
+    }
+}
+
+/// Converts the intermediate representation to a fixed-point number string and writes it to a file or stdout.
+///
+/// This function takes an `IntermediateRepresentation`, computes the scale factor based on
+/// the negative exponent, converts the mantissa to a `BigInt` and multiplies by the scale factor,
+/// constructs the fixed-point number string applying the sign, and writes the resulting string
+/// to the specified file or stdout.
+///
+/// # Arguments
+///
+/// * `inter_rep` - The intermediate representation to be converted.
+/// * `filepath_send` - A mutable reference to an optional `File` where the fixed-point
+///   number string will be written. If `None`, the result is written to stdout.
+///
+/// # Returns
+///
+/// This function returns a `std::io::Result<()>` which is `Ok` if the operation
+/// is successful, or an `Err` if an I/O error occurs while writing to the file.
+fn intermediate_to_fixed(
+    inter_rep: IntermediateRepresentation,
+    filepath_send: &mut Option<File>,
+) -> io::Result<()> {
```

Comment:
I haven't looked through this implementation carefully yet, but I would expect a "to_fixed" function to take the exponent of the fixed-point format as an argument.

Reply:
I think for this I am just using the exponent from the intermediate representation, does that make sense?